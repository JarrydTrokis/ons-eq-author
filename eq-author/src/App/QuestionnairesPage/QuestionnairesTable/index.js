import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { propType } from "graphql-anywhere";
import { TransitionGroup } from "react-transition-group";
import { isEmpty } from "lodash";
import gql from "graphql-tag";
import { colors } from "constants/theme";

import DeleteConfirmDialog from "components/DeleteConfirmDialog";
import questionConfirmationIcon from "./icon-questionnaire.svg";

import Row from "App/QuestionnairesPage/QuestionnairesTable/Row";

const Table = styled.table`
  width: 100%;
  font-size: 1em;
  border-collapse: collapse;
  table-layout: fixed;
  text-align: left;
`;

const TH = styled.th`
  padding: 1em;
  color: ${colors.darkGrey};
  width: ${props => props.colWidth};
  border-bottom: 1px solid #e2e2e2;
  font-weight: normal;
  font-size: 0.9em;
`;

TH.propTypes = {
  colWidth: PropTypes.string.isRequired,
};

const Panel = styled.div`
  background: white;
  border-radius: 4px;
  border: 1px solid #ccc;
`;

export class UnconnectedQuestionnairesTable extends React.PureComponent {
  static fragments = {
    QuestionnaireDetails: gql`
      fragment QuestionnaireDetails on Questionnaire {
        id
        title
        shortTitle
        displayName
        createdAt
        createdBy {
          id
          name
        }
      }
    `,
  };
  static propTypes = {
    questionnaires: PropTypes.arrayOf(
      propType(UnconnectedQuestionnairesTable.fragments.QuestionnaireDetails)
    ),
    onDeleteQuestionnaire: PropTypes.func.isRequired,
    onDuplicateQuestionnaire: PropTypes.func.isRequired,
  };

  state = {
    focusedId: null,
    showDeleteQuestionnaireDialog: false,
    deleteQuestionnaire: null,
  };

  handleOpenDeleteQuestionnaireDialog = deleteQuestionnaire => {
    this.setState({
      showDeleteQuestionnaireDialog: true,
      deleteQuestionnaire,
    });
  };

  handleCloseDeleteQuestionnaireDialog = () => {
    this.setState({
      showDeleteQuestionnaireDialog: false,
      deleteQuestionnaire: null,
    });
  };

  handleDeleteQuestionnaire = () => {
    const { questionnaires } = this.props;
    const { deleteQuestionnaire } = this.state;

    const possibleNextIndex =
      questionnaires.indexOf(
        questionnaires.find(q => q.id === deleteQuestionnaire.id)
      ) + 1;

    // If the last one is being removed then focus the one before that
    const nextIndex =
      possibleNextIndex > questionnaires.length - 1
        ? questionnaires.length - 2
        : possibleNextIndex;

    // We have to account to set focusedId to undefined when there are no
    // questionnaires left
    this.setState({
      focusedId: (questionnaires[nextIndex] || {}).id,
      showDeleteQuestionnaireDialog: false,
      deleteQuestionnaire: null,
    });

    this.props.onDeleteQuestionnaire(deleteQuestionnaire.id);
  };

  renderRows = questionnaires =>
    questionnaires.map(questionnaire => {
      return (
        <Row
          key={questionnaire.id}
          autoFocus={questionnaire.id === this.state.focusedId}
          questionnaire={questionnaire}
          onDeleteQuestionnaire={this.handleOpenDeleteQuestionnaireDialog}
          onDuplicateQuestionnaire={this.props.onDuplicateQuestionnaire}
        />
      );
    });

  render() {
    const { questionnaires } = this.props;
    const { showDeleteQuestionnaireDialog, deleteQuestionnaire } = this.state;

    if (isEmpty(questionnaires)) {
      return <p>You have no questionnaires</p>;
    }

    return (
      <Panel>
        <Table>
          <thead>
            <tr>
              <TH colWidth="50%">Title</TH>
              <TH colWidth="15%">Created</TH>
              <TH colWidth="20%">Owner</TH>
              <TH colWidth="15%">Actions</TH>
            </tr>
          </thead>
          <TransitionGroup component="tbody">
            {this.renderRows(questionnaires)}
          </TransitionGroup>
        </Table>
        <DeleteConfirmDialog
          isOpen={showDeleteQuestionnaireDialog}
          onClose={this.handleCloseDeleteQuestionnaireDialog}
          onDelete={this.handleDeleteQuestionnaire}
          title={deleteQuestionnaire && deleteQuestionnaire.displayName}
          alertText="This questionnaire including all sections and questions will be deleted."
          icon={questionConfirmationIcon}
          data-test="delete-questionnaire"
        />
      </Panel>
    );
  }
}

export default UnconnectedQuestionnairesTable;
