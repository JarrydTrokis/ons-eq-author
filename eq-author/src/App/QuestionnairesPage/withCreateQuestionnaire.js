import { graphql } from "react-apollo";
import createQuestionnaireQuery from "graphql/createQuestionnaire.graphql";
import { buildPagePath, buildIntroductionPath } from "utils/UrlUtils";

export const redirectToDesigner =
  (history) =>
  ({ data }) => {
    const questionnaire = data.createQuestionnaire;

    if (questionnaire.introduction) {
      history.push(
        buildIntroductionPath({
          questionnaireId: questionnaire.id,
          introductionId: questionnaire.introduction.id,
        })
      );
      return;
    }

    const section = questionnaire.sections[0];
    const page = section.folders[0].pages[0];

    history.push(
      buildPagePath({
        questionnaireId: questionnaire.id,
        pageId: page.id,
      })
    );
  };

export const mapMutateToProps = ({ ownProps, mutate }) => ({
  onCreateQuestionnaire: (questionnaire) => {
    const input = {
      ...questionnaire,
    };

    return mutate({
      variables: { input },
    }).then(redirectToDesigner(ownProps.history));
  },
});

export default graphql(createQuestionnaireQuery, {
  props: mapMutateToProps,
});
