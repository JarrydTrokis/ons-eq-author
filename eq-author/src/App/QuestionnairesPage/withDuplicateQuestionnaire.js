import { graphql } from "react-apollo";
import { get } from "lodash/fp";

import duplicateQuestionnaireMutation from "graphql/duplicateQuestionnaire.graphql";
import getQuestionnaireList from "graphql/getQuestionnaireList.graphql";

export const mapMutateToProps = ({ mutate }) => ({
  onDuplicateQuestionnaire({ id, displayName, createdBy }) {
    return mutate({
      variables: { input: { id } },
      optimisticResponse: {
        duplicateQuestionnaire: {
          id: `dupe-${id}`,
          displayName: `Copy of ${displayName}`,
          createdAt: new Date(Date.now()).toISOString(),
          createdBy,
          __typename: "Questionnaire",
        },
      },
    }).then(get("data.duplicateQuestionnaire"));
  },
});

export const handleUpdate = (proxy, { data: { duplicateQuestionnaire } }) => {
  const data = proxy.readQuery({ query: getQuestionnaireList });
  data.questionnaires = [duplicateQuestionnaire, ...data.questionnaires];
  proxy.writeQuery({
    query: getQuestionnaireList,
    data,
  });
};

export default graphql(duplicateQuestionnaireMutation, {
  props: mapMutateToProps,
  options: {
    update: handleUpdate,
  },
});
