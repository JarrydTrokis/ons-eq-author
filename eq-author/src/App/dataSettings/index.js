import React from "react";
import { Route, Redirect } from "react-router-dom";
import SampleFileDataPage from "./SampleFileData";
import SupplementaryDataPage from "./SupplementaryDataPage";
import { buildDataPath } from "utils/UrlUtils";

export default [
  <Route
    key="sample"
    exact
    path="/q/:questionnaireId/data/sample-file-data"
    component={SampleFileDataPage}
  />,
  <Route
    key="dataset"
    exact
    path="/q/:questionnaireId/data/supplementary-data"
    component={SupplementaryDataPage}
  />,
  <Route
    key="data"
    exact
    path="/q/:questionnaireId/data"
    render={(props) => (
      <Redirect to={`${buildDataPath(props.match.params)}/sample-file-data`} />
    )}
  />,
];
