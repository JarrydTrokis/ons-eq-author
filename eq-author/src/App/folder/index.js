import React from "react";

import { Route, Redirect } from "react-router-dom";
import { Query } from "react-apollo";
import SkipLogic from "App/shared/Logic/SkipLogic";

import Design from "./Design";
import RoutingPage from "./Routing";

import Error from "components/Error";

import GET_FOLDER_QUERY from "./graphql/getFolderQuery.graphql";

export default [
  <Route
    key="folder-design"
    path="/q/:questionnaireId/folder/:folderId/design"
    component={Design}
  />,
  <Route key="folder-logic" path="/q/:questionnaireId/folder/:folderId/logic">
    <Redirect to="routing" />
  </Route>,
  <Route
    key="folder-logic-routing"
    path="/q/:questionnaireId/folder/:folderId/routing"
    render={(props) => (
      <Query
        query={GET_FOLDER_QUERY}
        variables={{
          input: { folderId: props.match.params.folderId },
        }}
      >
        {({ loading, error, data }) => {
          if (loading) {
            return <React.Fragment />;
          }

          if (error) {
            return <Error>Error fetching folder</Error>;
          }

          if (data) {
            const { folder } = data;
            return folder.listId ? (
              <Redirect to="design" />
            ) : (
              <RoutingPage {...props} />
            );
          }
        }}
      </Query>
    )}
  />,
  <Route
    key="folder-logic-skip"
    path="/q/:questionnaireId/folder/:folderId/skip"
    render={(props) => (
      <Query
        query={GET_FOLDER_QUERY}
        variables={{
          input: { folderId: props.match.params.folderId },
        }}
      >
        {({ loading, error, data }) => {
          if (loading) {
            return <React.Fragment />;
          }

          if (error) {
            return <Error>Error fetching folder</Error>;
          }

          if (data) {
            const { folder } = data;
            return folder.listId ? (
              <Redirect to="design" />
            ) : (
              <SkipLogic {...props} />
            );
          }
        }}
      </Query>
    )}
  />,
];
