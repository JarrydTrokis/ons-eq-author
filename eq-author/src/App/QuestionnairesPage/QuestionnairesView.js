import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { propType } from "graphql-anywhere";
import styled from "styled-components";

import { isEmpty } from "lodash";

import QuestionnairesTable from "./QuestionnairesTable";

import { colors } from "constants/theme";

import Button from "components/buttons/Button";
import QuestionnaireSettingsModal from "App/QuestionnaireSettingsModal";
import flag from "./flag.svg";

const Header = styled.div`
  margin: 1em 0;
  display: flex;
  z-index: 1;
  align-items: center;
  justify-content: flex-end;
`;

const NoResults = styled.div`
  padding: 6em 0 10em;
  text-align: center;
`;

const NoResultsTitle = styled.h2`
  font-size: 1.3em;
  font-weight: 500;
  margin: 0 0 0.5em;

  &::before {
    content: url(${flag});
    display: block;
  }
`;

const NoResultsText = styled.p`
  margin: 0 0 1em;
  color: ${colors.textLight};
`;

class QuestionnairesView extends PureComponent {
  state = {
    isModalOpen: false,
  };

  handleModalOpen = () => this.setState({ isModalOpen: true });
  handleModalClose = () => this.setState({ isModalOpen: false });

  render() {
    const {
      questionnaires,
      onDeleteQuestionnaire,
      onDuplicateQuestionnaire,
      onCreateQuestionnaire,
    } = this.props;

    return (
      <>
        {isEmpty(questionnaires) ? (
          <NoResults>
            <NoResultsTitle>No questionnaires found</NoResultsTitle>
            <NoResultsText>
              {"You don't have access to any questionnaires yet."}
            </NoResultsText>
            <Button
              onClick={this.handleModalOpen}
              variant="primary"
              data-test="create-questionnaire"
            >
              Create a questionnaire
            </Button>
          </NoResults>
        ) : (
          <>
            <Header>
              <Button
                onClick={this.handleModalOpen}
                primary
                data-test="create-questionnaire"
              >
                Create questionnaire
              </Button>
            </Header>
            <QuestionnairesTable
              questionnaires={questionnaires}
              onDeleteQuestionnaire={onDeleteQuestionnaire}
              onDuplicateQuestionnaire={onDuplicateQuestionnaire}
            />
          </>
        )}
        <QuestionnaireSettingsModal
          isOpen={this.state.isModalOpen}
          onClose={this.handleModalClose}
          onSubmit={questionnaire => onCreateQuestionnaire(questionnaire)}
          confirmText="Create"
        />
      </>
    );
  }
}

QuestionnairesView.propTypes = {
  questionnaires: PropTypes.arrayOf(
    propType(QuestionnairesTable.fragments.QuestionnaireDetails)
  ),
  onCreateQuestionnaire: PropTypes.func.isRequired,
  onDeleteQuestionnaire: PropTypes.func.isRequired,
  onDuplicateQuestionnaire: PropTypes.func.isRequired,
};

export default QuestionnairesView;
