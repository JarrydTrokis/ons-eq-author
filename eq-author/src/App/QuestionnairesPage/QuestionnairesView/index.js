import React, { useState, useRef } from "react";
import PropTypes from "prop-types";
import { propType } from "graphql-anywhere";
import styled from "styled-components";
import { isEmpty } from "lodash";

import QuestionnaireSettingsModal from "App/QuestionnaireSettingsModal";
import Button from "components/buttons/Button";

import NoResults from "./NoResults";
import QuestionnairesTable from "./QuestionnairesTable";
import Footer from "./Footer";

import reducer, { buildInitialState, ACTIONS } from "./reducer";
import usePersistedReducer from "./usePersistedReducer";

export const STORAGE_KEY = "questionnaire-list-settings";
const STORED_KEYS = ["currentPageIndex"];

const Header = styled.div`
  margin: 1em 0;
  display: flex;
  z-index: 1;
  align-items: center;
  justify-content: flex-end;
`;

const QuestionnairesView = ({
  questionnaires,
  onDeleteQuestionnaire,
  onDuplicateQuestionnaire,
  onCreateQuestionnaire,
}) => {
  const [isModalOpen, setIsModalOpen] = useState(false);

  const handleModalOpen = () => setIsModalOpen(true);
  const handleModalClose = () => setIsModalOpen(false);

  const questionnairesRef = useRef(questionnaires);

  const [state, dispatch] = usePersistedReducer(
    STORAGE_KEY,
    STORED_KEYS,
    reducer,
    { currentPageIndex: 0 },
    buildInitialState(questionnairesRef.current)
  );

  if (questionnaires !== questionnairesRef.current) {
    dispatch({
      type: ACTIONS.SET_QUESTIONNAIRES,
      payload: questionnaires,
    });
    questionnairesRef.current = questionnaires;
  }

  const handleDeleteQuestionnaire = questionnaire => {
    dispatch({
      type: ACTIONS.DELETE_QUESTIONNAIRE,
      payload: questionnaire,
    });
    onDeleteQuestionnaire(questionnaire.id);
  };

  if (isEmpty(state.questionnaires)) {
    return (
      <>
        <NoResults onCreateQuestionnaire={handleModalOpen} />
        <QuestionnaireSettingsModal
          isOpen={isModalOpen}
          onClose={handleModalClose}
          onSubmit={onCreateQuestionnaire}
          confirmText="Create"
        />
      </>
    );
  }

  return (
    <>
      <Header>
        <Button
          onClick={handleModalOpen}
          primary
          data-test="create-questionnaire"
        >
          Create questionnaire
        </Button>
      </Header>
      <QuestionnairesTable
        questionnaires={state.currentPage}
        onDeleteQuestionnaire={handleDeleteQuestionnaire}
        onDuplicateQuestionnaire={onDuplicateQuestionnaire}
        autoFocusId={state.autoFocusId}
      />
      <Footer
        countOnPage={state.currentPage.length}
        totalCount={state.questionnaires.length}
        pageCount={state.pages.length}
        currentPageIndex={state.currentPageIndex}
        onPageChange={newPage =>
          dispatch({ type: ACTIONS.CHANGE_PAGE, payload: newPage })
        }
      />
      <QuestionnaireSettingsModal
        isOpen={isModalOpen}
        onClose={handleModalClose}
        onSubmit={onCreateQuestionnaire}
        confirmText="Create"
      />
    </>
  );
};

QuestionnairesView.propTypes = {
  questionnaires: PropTypes.arrayOf(
    propType(QuestionnairesTable.fragments.QuestionnaireDetails)
  ),
  onCreateQuestionnaire: PropTypes.func.isRequired,
  onDeleteQuestionnaire: PropTypes.func.isRequired,
  onDuplicateQuestionnaire: PropTypes.func.isRequired,
};

export default QuestionnairesView;
