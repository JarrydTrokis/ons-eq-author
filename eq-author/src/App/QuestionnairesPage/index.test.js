import React from "react";
import { UnconnectedQuestionnairesPage } from "App/QuestionnairesPage";
import { shallow } from "enzyme";

import { READ } from "constants/questionnaire-permissions";

describe("components/QuestionnairesPage", () => {
  const createWrapper = props =>
    shallow(
      <UnconnectedQuestionnairesPage
        onDeleteQuestionnaire={jest.fn()}
        onCreateQuestionnaire={jest.fn()}
        onDuplicateQuestionnaire={jest.fn()}
        {...props}
      />
    );

  it("should not render table whilst data is loading", () => {
    const wrapper = createWrapper({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it("should render loading message when loading", () => {
    const instance = createWrapper().instance();
    const renderResults = instance.renderResults({
      loading: true,
    });

    expect(renderResults).toMatchSnapshot();
  });

  it("should render when there are no questionnaires", () => {
    const wrapper = createWrapper({ questionnaires: [] });
    expect(wrapper).toMatchSnapshot();
  });

  it("should render error message when there is an error", () => {
    const instance = createWrapper().instance();
    const renderResults = instance.renderResults({
      error: true,
    });

    expect(renderResults).toMatchSnapshot();
  });

  it("should render table when there are questionnaires", () => {
    const questionnaires = [
      {
        id: "1",
        displayName: "Test questionnaire",
        title: "Test questionnaire",
        shortTitle: "Short title",
        createdAt: "01/01/1970",
        updatedAt: "02/01/1970",
        createdBy: {
          name: "Mike",
          id: "1",
          email: "mike@mail.com",
          displayName: "Mike",
        },
        sections: [
          {
            id: "5",
            pages: [{ id: "10" }],
          },
        ],
        permission: READ,
      },
    ];

    const me = {
      id: "1",
      name: "Test",
      email: "test@email.com",
      displayName: "test",
    };

    const wrapper = createWrapper({ questionnaires });
    const instance = wrapper.instance();
    const renderResults = instance.renderResults({
      data: { questionnaires, me },
    });

    expect(wrapper).toMatchSnapshot();
    expect(renderResults).toMatchSnapshot();
  });

  it("should correctly render title", () => {
    const instance = createWrapper().instance();
    const renderTitle = instance.renderTitle("FooBar");

    expect(renderTitle).toMatchSnapshot();
  });
});
