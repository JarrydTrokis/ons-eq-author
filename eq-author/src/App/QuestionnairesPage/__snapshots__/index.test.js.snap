// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/QuestionnairesPage should correctly render title 1`] = `"Your Questionnaires - FooBar"`;

exports[`components/QuestionnairesPage should not render table whilst data is loading 1`] = `
<GetContext
  title={[Function]}
>
  <BaseLayout
    title="Your Questionnaires"
  >
    <ScrollPane
      permanentScrollBar={true}
    >
      <MainCanvas
        maxWidth="55em"
      >
        <Query
          fetchPolicy="cache-and-network"
          query={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GetQuestionnaireList",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "questionnaires",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "directives": Array [],
                              "kind": "FragmentSpread",
                              "name": Object {
                                "kind": "Name",
                                "value": "QuestionnaireDetails",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
                Object {
                  "directives": Array [],
                  "kind": "FragmentDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "QuestionnaireDetails",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "title",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "shortTitle",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "displayName",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "createdAt",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "updatedAt",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "createdBy",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "typeCondition": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "Questionnaire",
                    },
                  },
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 296,
                "start": 0,
              },
            }
          }
        >
          <Component />
        </Query>
      </MainCanvas>
    </ScrollPane>
  </BaseLayout>
</GetContext>
`;

exports[`components/QuestionnairesPage should render error message when there is an error 1`] = `
<Error>
  Oops! Questionnaires could not be found
</Error>
`;

exports[`components/QuestionnairesPage should render loading message when loading 1`] = `
<Loading
  height="24.25rem"
>
  Questionnaires loadingâ€¦
</Loading>
`;

exports[`components/QuestionnairesPage should render table when there are questionnaires 1`] = `
<GetContext
  title={[Function]}
>
  <BaseLayout
    title="Your Questionnaires"
  >
    <ScrollPane
      permanentScrollBar={true}
    >
      <MainCanvas
        maxWidth="55em"
      >
        <Query
          fetchPolicy="cache-and-network"
          query={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GetQuestionnaireList",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "questionnaires",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "directives": Array [],
                              "kind": "FragmentSpread",
                              "name": Object {
                                "kind": "Name",
                                "value": "QuestionnaireDetails",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
                Object {
                  "directives": Array [],
                  "kind": "FragmentDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "QuestionnaireDetails",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "title",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "shortTitle",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "displayName",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "createdAt",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "updatedAt",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "createdBy",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "typeCondition": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "Questionnaire",
                    },
                  },
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 296,
                "start": 0,
              },
            }
          }
        >
          <Component />
        </Query>
      </MainCanvas>
    </ScrollPane>
  </BaseLayout>
</GetContext>
`;

exports[`components/QuestionnairesPage should render table when there are questionnaires 2`] = `
<QuestionnairesView
  onCreateQuestionnaire={[MockFunction]}
  onDeleteQuestionnaire={[MockFunction]}
  onDuplicateQuestionnaire={[MockFunction]}
  questionnaires={
    Array [
      Object {
        "createdAt": "01/01/1970",
        "createdBy": Object {
          "id": "1",
          "name": "Mike",
        },
        "displayName": "Test questionnaire",
        "id": "1",
        "sections": Array [
          Object {
            "id": "5",
            "pages": Array [
              Object {
                "id": "10",
              },
            ],
          },
        ],
        "shortTitle": "Short title",
        "title": "Test questionnaire",
        "updatedAt": "02/01/1970",
      },
    ]
  }
/>
`;

exports[`components/QuestionnairesPage should render when there are no questionnaires 1`] = `
<GetContext
  title={[Function]}
>
  <BaseLayout
    title="Your Questionnaires"
  >
    <ScrollPane
      permanentScrollBar={true}
    >
      <MainCanvas
        maxWidth="55em"
      >
        <Query
          fetchPolicy="cache-and-network"
          query={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GetQuestionnaireList",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "questionnaires",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "directives": Array [],
                              "kind": "FragmentSpread",
                              "name": Object {
                                "kind": "Name",
                                "value": "QuestionnaireDetails",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
                Object {
                  "directives": Array [],
                  "kind": "FragmentDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "QuestionnaireDetails",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "title",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "shortTitle",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "displayName",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "createdAt",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "updatedAt",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "createdBy",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "typeCondition": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "Questionnaire",
                    },
                  },
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 296,
                "start": 0,
              },
            }
          }
        >
          <Component />
        </Query>
      </MainCanvas>
    </ScrollPane>
  </BaseLayout>
</GetContext>
`;
